.github/workflows/check-and-test.yml

```diff
@@ -63,6 +63,37 @@ jobs:
       - name: Run chart-testing (lint)
         run: ct lint --validate-maintainers false --charts helm/dscp-node/
 
+  check-weights-have-changes:
+    name: Check Weights
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+      pull-requests: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: check
+        id: check
+        run: |
+          set -x
+          URL="https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.pull_request.number }}/files"
+          files_changed_data=$(curl -s --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' -X GET -G "$URL")
+          files_changed="$(echo $files_changed_data | jq -r '.[] | .filename')"
+          # Adding || true to avoid "Process exited with code 1" errors
+          pallet_dirs_changed="$(echo "${files_changed}" | xargs dirname | grep -v "pallets/traits/*" | grep -o "pallets/[^/]*" | sort | uniq || true)"
```

Comment:
as well as ignoring `traits`, I think `doas` and `transaction-payment-free` can be ignored as they don't have `weights.rs`